paddle/fluid/framework/details/fetch_op_handle.cc

```diff
@@ -62,7 +63,8 @@ void FetchOpHandle::RunImpl() {
     auto &t = var->Get<framework::LoDTensor>();
     if (platform::is_gpu_place(t.place())) {
 #ifdef PADDLE_WITH_CUDA
-      TensorCopySync(t, cpu, &tensors_[i]);
+      TensorCopy(t, cpu, *dev_ctxes_.at(t.place()), &tensors_[i]);
+      dev_ctxes_.at(t.place())->Wait();
```

Comment:
Don't use default stream here, it may make program slow. https://devblogs.nvidia.com/gpu-pro-tip-cuda-7-streams-simplify-concurrency/